GeoMesaQuery.md

```diff
@@ -0,0 +1,81 @@
+# GeoMesa Indexing and Querying #
+
+## The GeoMesa Index ##
```

Comment:
Z3 - spatio-temporal (lat/lon/time):
* row
 * 1 byte for the shard, based on hash of the feature ID, to distribute load across tservers
 * 2 bytes for the 'short' containing weeks since the java epoch (1970)
   * note: this can be configured to be days/weeks/months/years
 * 8 bytes for the 'long' containing the encoded z value - 21 bits lat, 21 bits lon, 20 bits time - for non-points this is truncated to 3 bytes
   * note: time will be milliseconds in the current day, second in the current week/month, minutes in the current year
 * n bytes for the feature ID
* column family - constant 'F'
* column qualifier - empty
* value - serialized simple feature

Z2 - spatial (lat/lon)
* row
 * 1 byte identifying the simple feature type (to allow multiple types in a single table - optional - table sharing can be turned on or off)
 * 1 byte for a shard
 * 8 bytes for the 'long' encoding the z value - 31 bits lat, 31 bits lon - for non-points this is truncated down to 3 bytes
 * n bytes for the feature ID
* column family - constant 'F'
* column qualfier - empty
* value - serialized simple feature

Record - ID lookup
* row
 * 1 byte identifying the simple feature type (optional - table sharing can be turned on or off)
 * n bytes for the feature ID
* column family - constant 'F'
* column qualifier - empty
* value - serialized simple feature

Attribute - attribute value index
* row
 * 1 byte identifying the simple feature type (optional - table sharing can be turned on or off)
 * 2 bytes identifying the attribute being indexed
 * n bytes storing the lexicoded attribute value
 * NULLBYTE (0) as a separator
 * 12 bytes storing the date of the feature (optional - only if the schema has a date field)
 * n bytes storing the feature ID
* column family -  constant 'F'
* column qualifier - empty
* value - serialized simple feature, or serialized geometry and date (can be configured for either)